Src/FluentAssertions/Numeric/NumericAssertions.cs

```diff
@@ -60,7 +63,7 @@ public AndConstraint<TAssertions> Be(T expected, string because = "", params obj
             Execute.Assertion
                 .ForCondition(Subject.HasValue && Subject.Value.CompareTo(expected) == 0)
                 .BecauseOf(because, becauseArgs)
-                .FailWith("Expected {context:value} to be {0}{reason}, but found {1}.", expected, Subject);
+                .FailWith("Expected {context:value} to be {0}{reason}, but found {1}" + GenerateDifferenceMessage(expected), expected, Subject);
```

Comment:
Would be happy to find an alternative to this string concatenation.

I tried using `.FailWith("Expected {context:value} to be {0}{reason}, but found {1}{2}.", expected, Subject, GenerateDifferenceMessage(expected)`
but the `MessageBuilder.FormatArgumentPlaceholders` added the string along with its double quotes:
`"Expected value to be 2 because we want to test the failure message, but found 1" (which differs by -1)."."` 

Any suggestions on how to make it cleaner?

Reply:
I could possibly create a `NumericComparisonResult<T>` class that will hold the `CalculateDifference` result and add that as an argument to `FailWith`.
Then create a dedicated `NumericComparisonResultFormatter` which will hold the logic that currently resides in `GenerateDifferenceMessage`.

Does that make sense?